src/amazon-qindex-mcp-server/awslabs/amazon_qindex_mcp_server/server.py

```diff
@@ -0,0 +1,422 @@
+                        {
+                            'name': 'string',
+                            'value': {
+                                # Various value types based on attribute
+                            }
+                        }
+                    ],
+                    'documentId': 'string',
+                    'documentTitle': 'string',
+                    'documentUri': 'string',
+                    'scoreAttributes': {
+                        'scoreConfidence': 'string'
+                    }
+                }
+            ]
+        }
+
+    Raises:
+        ValueError: If there's an error with the Q Business API call or if credentials are missing/invalid
+    """
+    try:
+        # Create QBusinessClient with provided credentials
+        client = QBusinessClient(
+            region_name=qbuiness_region,
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            aws_session_token=aws_session_token,
+        )
+
+        # Convert models to dictionaries
+        content_source_dict = None
+        if content_source:
+            content_source_dict = content_source.model_dump(exclude_none=True)
+            if 'retriever' in content_source_dict:
+                content_source_dict = {'retriever': content_source_dict['retriever']}
+
+        attribute_filter_dict = None
+        if attribute_filter:
+            attribute_filter_dict = attribute_filter.model_dump(exclude_none=True)
+
+        # Ensure max_results is properly typed
+        max_results_int = int(max_results) if max_results is not None else None
+
+        # Perform the search
+        return client.search_relevant_content(
+            application_id=str(application_id),
+            query_text=str(query_text),
+            attribute_filter=attribute_filter_dict,
+            content_source=content_source_dict,
+            max_results=max_results_int,
+            next_token=str(next_token) if next_token else None,
+        )
+    except Exception as e:
+        logger.error(f'Error searching Q Business content: {str(e)}')
+        if not aws_access_key_id or not aws_secret_access_key or not aws_session_token:
+            raise ValueError(
+                'Missing AWS credentials. Please follow the authentication flow described in the tool documentation.'
+            )
+        raise ValueError(str(e))
+
+
+def main():
+    """Run the MCP server with CLI argument support."""
+    parser = argparse.ArgumentParser(
+        description='An AWS Labs Model Context Protocol (MCP) server for amazon_qindex'
+    )
+    parser.add_argument('--sse', action='store_true', help='Use SSE transport')
+    parser.add_argument('--port', type=int, default=8888, help='Port to run the server on')
+
+    args = parser.parse_args()
+
+    # Run server with appropriate transport
+    if args.sse:
+        mcp.settings.port = args.port
+        mcp.run(transport='sse')
+
+    else:
+        mcp.run()
```

Comment:
we are not supporting `sse` in this repo for security, please update this: 
```suggestion
   mcp.run()
```